### PR TITLE
Feature/aleph provider refactoring

### PR DESCRIPTION
**Pull request by tripox**

### CODE REVIEW

File: modules/aleph/includes/aleph.user_consent.inc [hunk 1]
```diff
@@ -28,59 +28,43 @@ function aleph_user_consent_info() {
...
 /**
  * Retrieve any consent record which the user has saved.
  *
- * @param object $account
- *   The user account.
- *
  * @return array
  *   An array of consent arrays sorted by the date they were accepted.
  */
-function aleph_user_consent_get_consent($account) {
```
Comment (xendk) [inline modules/aleph/includes/aleph.user_consent.inc]: Why? They still get called with parameters, so removing them can cause notices.

File: modules/aleph/src/Aleph/AlephPatron.php [hunk 2]
```diff
@@ -30,18 +32,45 @@ public function __construct(AlephClient $client, $borId, $verification) {
...
     }
+
+    return FALSE;
+  }
+
+  /**
+   * Check if patron is authenticated.
+   */
+  public function isAuthenticated() {
+    $this->authenticated ? $authenticated = TRUE : $authenticated = FALSE;
```
Comment (xendk) [inline modules/aleph/src/Aleph/AlephPatron.php]: What's wrong with `return (bool) $this->authenticated;`?

**Review by xendk — APPROVED** [review]
Approved with minor niggles.

File: modules/aleph/includes/aleph.user_consent.inc (hunk 1 above)
Comment (tripox) [reply modules/aleph/includes/aleph.user_consent.inc]: You're right. PHPStorm was bitching about it, so I removed it without thinking. :)

File: modules/aleph/src/Aleph/AlephPatron.php (hunk 2 above)
Comment (tripox) [reply modules/aleph/src/Aleph/AlephPatron.php]: Duh. Fixed. :)

**Pull request merged**